src/StreamJsonRpc/MessagePackFormatter.cs

```diff
@@ -587,6 +587,9 @@ private MessagePackSerializerOptions MassageUserDataOptions(MessagePackSerialize
             // Add our own resolvers to fill in specialized behavior if the user doesn't provide/override it by their own resolver.
             var resolvers = new IFormatterResolver[]
             {
+                // Support for marshalled objects.
+                new RpcMarshalableImplicitResolver(this),
```

Comment:
If I switch the order of the resolvers, an interface that is both marshalable and serializable (e.g. through `MessagePackFormatterAttribute`) would be serialized:
1. it would be a different behavior than when using Newtonsoft.Json
2. we discussed in [this comment](https://github.com/microsoft/vs-streamjsonrpc/pull/777#discussion_r825001683) that marshalability should have precedence. For reference:
> `RpcMarshable` should win because it's universal (same attribute regardless of formatter), and switching formatters shouldn't change a type from serializing to marshalling

Reply:
I get that if you move it too far down it would be serialized, but the user-types resolver is typically empty until they populate it IIRC. So I thought one position down wouldn't have broken this.

Reply:
Moving it one position down breaks the case I described above because `userSuppliedOptions.Resolver` by default includes `AttributeFormatterResolver`.

Reply:
Ok. We'll leave it then.